cmd/amass/main.go

```diff
@@ -45,492 +33,93 @@ var (
-	list          = flag.Bool("list", false, "Print the names of all available data sources")
-	vprint        = flag.Bool("version", false, "Print the version number of this Amass binary")
-	dir           = flag.String("dir", "", "Path to the directory containing the output files")
-	config        = flag.String("config", "", "Path to the INI configuration file. Additional details below")
-	maxdns        = flag.Int("max-dns-queries", 0, "Maximum number of concurrent DNS queries")
-	unresolved    = flag.Bool("include-unresolvable", false, "Output DNS names that did not resolve")
-	ips           = flag.Bool("ip", false, "Show the IP addresses for discovered names")
-	ipv4          = flag.Bool("ipv4", false, "Show the IPv4 addresses for discovered names")
-	ipv6          = flag.Bool("ipv6", false, "Show the IPv6 addresses for discovered names")
-	brute         = flag.Bool("brute", false, "Execute brute forcing after searches")
-	active        = flag.Bool("active", false, "Attempt zone transfers and certificate name grabs")
-	norecursive   = flag.Bool("norecursive", false, "Turn off recursive brute forcing")
-	minrecursive  = flag.Int("min-for-recursive", 0, "Number of subdomain discoveries before recursive brute forcing")
-	passive       = flag.Bool("passive", false, "Disable DNS resolution of names and dependent features")
-	noalts        = flag.Bool("noalts", false, "Disable generation of altered names")
-	sources       = flag.Bool("src", false, "Print data sources for the discovered names")
-	wordlist      = flag.String("w", "", "Path to a different wordlist file")
-	altwordlist   = flag.String("aw", "", "Path to a different wordlist file for alterations")
-	allpath       = flag.String("oA", "", "Path prefix used for naming all output files")
-	logpath       = flag.String("log", "", "Path to the log file where errors will be written")
-	outpath       = flag.String("o", "", "Path to the text output file")
-	jsonpath      = flag.String("json", "", "Path to the JSON output file")
-	datapath      = flag.String("do", "", "Path to data operations output file")
-	namespath     = flag.String("nf", "", "Path to a file providing already known subdomain names")
-	domainspath   = flag.String("df", "", "Path to a file providing root domain names")
-	excludepath   = flag.String("ef", "", "Path to a file providing data sources to exclude")
-	includepath   = flag.String("if", "", "Path to a file providing data sources to include")
-	resolvepath   = flag.String("rf", "", "Path to a file providing preferred DNS resolvers")
-	blacklistpath = flag.String("blf", "", "Path to a file providing blacklisted subdomains")
 )
 
+func commandUsage(msg string, cmdFlagSet *flag.FlagSet, errBuf *bytes.Buffer) {
+	amass.PrintBanner()
+	g.Fprintf(color.Error, "Usage: %s %s\n\n", path.Base(os.Args[0]), msg)
+	cmdFlagSet.PrintDefaults()
+	g.Fprintln(color.Error, errBuf.String())
+	g.Fprintf(color.Error, "An example configuration file can be found here: \n%s\n\n", exampleConfigFileURL)
```

Comment:
So it's unclear what each subcommand can use from the config file. What about having an example config file for each subcommand that shows all the options supported by each subcommand?

Reply:
I am adding the subcommand specific info into the user guide documentation, and will add a link to the user guide to the help information